packages/storage/amplify_storage_s3_dart/lib/src/exception/s3_storage_exception.dart

```diff
@@ -12,228 +11,69 @@ const _keyNotFoundRecoveryMessage =
 const _httpErrorRecoveryMessage =
     'HTTP error returned from service, please review the `underlyingException` for details.';
 
-/// {@template amplify_storage_s3_dart.s3_storage_exception}
-/// Represents exceptions that may be thrown calling Storage S3 plugin APIs.
-/// {@endtemplate}
-class S3Exception extends StorageException {
-  /// {@macro amplify_storage_s3_dart.s3_storage_exception}
-  const S3Exception(
-    super.message, {
-    required String super.recoverySuggestion,
-    super.underlyingException,
-  });
-
-  /// Creates a [S3Exception] from a FileSystemException.
-  factory S3Exception.fromFileSystemException(Exception exception) {
-    return S3Exception(
-      'Error occurred while reading/writing file in the local file system.',
-      recoverySuggestion: 'Please review the underlying exception.',
-      underlyingException: exception,
-    );
-  }
-
-  /// Creates a [S3Exception] that represents an error that shouldn't
-  /// happen normally.
-  factory S3Exception.unknownException([String? extraMessage]) {
-    return S3Exception(
-      'Unknown exception occurred. $extraMessage',
-      recoverySuggestion: AmplifyExceptionMessages.missingExceptionMessage,
-    );
-  }
-
-  /// Creates a [S3Exception] that represents an unexpected error
-  /// when download operation doesn't receive all bytes when download completes.
-  factory S3Exception.incompleteDownload() {
-    return const S3Exception(
-      'Download has completed, but it has not received all bytes of content.',
-      recoverySuggestion: AmplifyExceptionMessages.missingExceptionMessage,
+/// The exception thrown when cancel a controllable operation.
+const s3ControllableOperationCanceledException = StorageOperationCanceled(
+  'The operation has been canceled.',
+  recoverySuggestion:
+      'This is an expected exception when you call cancel() API on the storage operation object. You need to handle this exception to take further action on operation cancellation.',
+);
+
+/// Creates a [StorageKeyNotFoundException] with the [underlyingException].
+StorageKeyNotFoundException createKeyNotFoundException(
+  Exception underlyingException,
+) =>
+    StorageKeyNotFoundException(
+      'Key is not found.',
```

Comment:
Should we specify what specific key this is? 

Such as specifying this is the key for the storage item, like:

Cannot find key for the storage item. 

The storage item specified with the provided key doesn't exist.